source/common/router/upstream_request.cc

```diff
@@ -598,9 +598,6 @@ void UpstreamRequest::onPoolReady(std::unique_ptr<GenericUpstream>&& upstream,
   had_upstream_ = true;
   // Have the upstream use the account of the downstream.
   upstream_->setAccount(parent_.callbacks()->account());
-  if (enable_half_close_) {
```

Comment:
The `enable_half_close_` value is plumbed through the `FilterManagerCallbacks::isHalfCloseEnabled()` to the filter manager, which determines if `local_complete_` is set on encoder end_stream for the upstream filter manager.

The `GenericUpstream::enableHalfClose()` is setting the TCP proxying mode in H/1 codec which controls how codec behaves when it sees decoder end stream. This code used the half-close flag here to also set the TCP proxying mode, which worked before since this was the half-close mode was only used in TCP proxy mode.

In #34461 I'm going to enable half-close mode in other cases, so the TCP proxy settings needs to be separate from half-close. So I have moved setting the TCP proxy mode to the dedicated `RouterUpstreamRequest` class for TCP proxy upstreams.

I have updated PR description accordingly.